tools/shared.py

```diff
@@ -772,6 +772,32 @@ def clean_temp():
     prepare_to_clean_temp(EMSCRIPTEN_TEMP_DIR) # this global var might change later
   return EMSCRIPTEN_TEMP_DIR
 
+class WarningManager:
+  WARN_ABSOLUTE_PATHS = False # by default, do not warn about absolute paths.
+  WARN_SEPARATE_ASM = True  # by default user is warned that they must load the separate 'asm.js' if html output is not used
+  WARN_ALMOST_ASM = True # disable this if you are expecting "almost asm" output, e.g. ALLOW_MEMORY_GROWTH=1 or SPLIT_MEMORY=1
+
+  @staticmethod
+  def capture_warnings(cmd_args):
+    assert cmd_args
+
+    for i in range(len(cmd_args)):
+      if cmd_args[i] == '-Wwarn-absolute-paths':
+        cmd_args[i] = ''
+        WarningManager.WARN_ABSOLUTE_PATHS = True
+      elif cmd_args[i] == '-Wno-warn-absolute-paths':
+        cmd_args[i] = ''
+        WarningManager.WARN_ABSOLUTE_PATHS = False
+      elif cmd_args[i] == '-Wno-warn-separate-asm':
```

Comment:
We should support both the `yes` and `no` forms of each of the flags, as you did with `absolute-paths`. Rather than explicit cases for every possible flag, it might make more sense to just see if the flag starts with "-W" and then see if it next has "no-" and then switch on the rest of the string. Actually you wouldn't even need to do  a switch or cases. You could just store the string in a dict with a boolean saying whether it was enabled or disabled and look that up again when `warn` is called. We would still want a dict containing all of the known flags, so that any flags with unknown warning options can be passed through to the underlying clang.